### ISSUE TITLE
[BUG]: Doesn't work as expected

### ISSUE THREAD

**Issue by deanbirnie** [wasi-master/13ft#14]
### Description of the bug

At first I tried it on a local News site and got blocked by Cloudflare. So I thought I'd use a Medium article and got the same blocked by Cloudflare page. 

[Link 1](https://www.news24.com/fin24/climate_future/energy/long-awaited-energy-plan-drops-brace-for-potentially-more-load-shedding-for-longer-20240105)
[Link 2](https://amandaclaypool.medium.com/what-spotifys-layoffs-tell-us-about-the-future-of-work-b7f080268d08)

Do you have your instance running as per your GitHub ReadMe? 

I have installed this as a Docker app on my unRAID server. Will play around with it a bit more and see if I can figure anthing out.

### Steps To Reproduce

Use links above and see if they can be accessed.

**Comment by evilalmus** [comment]
The same thing for me. The error says "Please enable cookies"

**Comment by wasi-master** [comment]
Yeah, I've yet not found a fix for this issue. If anyone wants to help or has any ideas please say.

**Comment by oggiedoggie62** [comment]
Mine worked as long as I include the http:// in the domain name. But it doesnt block ads like 12ft.io . so it does have the benefit of being self hosted but does not block ads but so far plage load for me
OK yeah medium did block me :(

**Comment by JPDVM2014** [comment]
I always get the Cloudflare block too.

**Comment by 33b5e5** [comment]
When trying to access WSJ, I get "Please enable JS and disable any ad blocker." Links from The Atlantic load, but style sheets are broken, which makes it difficult to read. Other news sites with paywalls also didn’t work well; styles are stripped and/or the content doesn't expand as expected. These all seem to work on archive.today for comparison.

**Comment by cheesestringer** [comment]
The larger sites are probably verifying the crawler https://developers.google.com/search/docs/crawling-indexing/verifying-googlebot

**Comment by wasi-master** [comment]
Medium (link 2) works now as of https://github.com/wasi-master/13ft/pull/43

**Comment by wasi-master** [comment]
Both links in this issue now resolve:

- **news24.com** (link 1) — fixed in commit 9069a1c; the article renders fully (verified: 28 paragraphs, no Cloudflare/cookie challenge).
- **Medium** (link 2) — fixed via #43 (Freedium fallback), as already noted above.

Closing as resolved. If a paywalled site still fails for you, please open a new issue with the specific URL — the failure modes differ per site (Cloudflare vs DataDome vs metered paywall) and are best tracked separately.
